blog/views.py

```diff
@@ -0,0 +1,60 @@
+from django.shortcuts import render, redirect
+from django.http import HttpResponse, HttpRequest
+from django.contrib.auth.decorators import login_required
+from django.views import generic
+from django.urls import reverse_lazy
+from django.core.paginator import Paginator
+from django.contrib.auth.mixins import LoginRequiredMixin
+from .forms import CommentaryForm
+
+from .models import Post
+
+
+def index(request: HttpRequest) -> HttpResponse:
+    posts = (
+        Post.objects.all()
+        .prefetch_related("commentarys")
+        .order_by("-created_time")
+    )
+    for post in posts:
+        post.content = post.content[:80] + "..."
+    paginator = Paginator(posts, 5)
+    page_number = request.GET.get("page")
+    page_obj = paginator.get_page(page_number)
+    context = {
+        "posts": page_obj.object_list,
+        "post_list": page_obj,
+        "paginator": paginator,
+    }
+    return render(request, "blog/index.html", context=context)
+
+
+class PostDetailView(generic.DetailView):
+    model = Post
+    queryset = Post.objects.all().prefetch_related("commentarys")
+    template_name = "blog/post_detail.html"
+
+
+class PostCreateView(LoginRequiredMixin, generic.CreateView):
+    model = Post
+    fields = "__all__"
+    success_url = reverse_lazy("blog:post-detail")
+    template_name = "blog/post_form.html"
+
+
+class CommentaryCreateView(LoginRequiredMixin, generic.CreateView):
+    model = Post
```

Comment:
The `model` for `CommentaryCreateView` should be `Commentary` instead of `Post`. This is necessary to ensure the form correctly handles commentary data.

Reply:
No, i create comment in the CommentaryForm